javascripts/govuk/analytics/tracker.js

```diff
@@ -0,0 +1,81 @@
+(function() {
+  "use strict";
+  window.GOVUK = window.GOVUK || {};
+
+  var Tracker = function(config) {
+    var setDimension = this.setDimension.bind(this);
+
+    this.universal = new GOVUK.GoogleAnalyticsUniversalTracker(config.universalId, config.cookieDomain);
+    this.classic = new GOVUK.GoogleAnalyticsClassicTracker(config.classicId, config.cookieDomain);
+
+    setPixelDensityDimension();
+    setHTTPStatusCodeDimension();
+
+    if (config.prePageviewConfiguration && typeof config.prePageviewConfiguration === 'function') {
+      config.prePageviewConfiguration();
+    }
+
+    this.trackPageview();
+
+    function setPixelDensityDimension() {
+      var pixelRatioDimensionIndex = 11;
+
+      if (window.devicePixelRatio) {
+        setDimension(pixelRatioDimensionIndex, window.devicePixelRatio, 'Pixel Ratio', 2);
+      }
+    }
+
+    function setHTTPStatusCodeDimension() {
```

Comment:
I'm wondering whether `setPixelDensityDimension` and `setHTTPStatusCodeDimension` can be omitted here, and setup using the new `prePageviewConfiguration` by apps that need it. It would make the tracker more generic and avoid the need for `.bind` on `setDimension`.